components/product-carousel/product-carousel.tsx

```diff
@@ -0,0 +1,119 @@
+import { removeEdgesAndNodes } from '@bigcommerce/catalyst-client';
+import { useQuery } from '@tanstack/react-query';
+import { forwardRef, useDeferredValue, useId } from 'react';
+
+import {
+  Carousel,
+  CarouselContent,
+  CarouselNextIndicator,
+  CarouselPreviousIndicator,
+  CarouselSlide,
+} from '@bigcommerce/components/carousel';
+
+import { getQuickSearchResults } from '../../client/queries/get-quick-search-results';
+
+import { DummyCard } from './dummy-card';
+import { Pagination } from './pagination';
+import { Product, ProductCard } from './product-card';
+
+interface Props {
+  title: string;
+  categoryId?: number;
+}
+
+type QuickSearchResults = Awaited<ReturnType<typeof getQuickSearchResults>>;
+
+export const fetchQuickSearch = async ({ categoryId }: { categoryId?: number }) => {
+  const response = await fetch(
+    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+    `/api/quick-search?categoryId=${categoryId!}&imageHeight=500&imageWidth=500&limit=10`,
+  );
+
+  // eslint-disable-next-line @typescript-eslint/consistent-type-assertions
+  const { products } = (await response.json()) as QuickSearchResults;
```

Comment:
Is it feasible to create a zod schema here and parse this?

Reply:
Yeah... but I was a little lazy here and just followed what was going on in this file: https://github.com/makeswift/catalyst-demo/blob/main/components/cart/client.tsx#L15